domain_config_ui/src/Config/ConfigFactory.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace Drupal\domain_config_ui\Config;
+
+use Drupal\Core\Config\ConfigFactory as CoreConfigFactory;
+use Drupal\Core\Config\StorageInterface;
+use Drupal\Core\Config\TypedConfigManagerInterface;
+use Drupal\domain_config_ui\DomainConfigUIManager;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Extends core ConfigFactory class to save domain specific configuration.
+ */
+class ConfigFactory extends CoreConfigFactory {
+
+  /**
+   * List of config that should always be saved globally. Use * for wildcards.
+   *
+   * @TODO: Document the logic of this list.
+   *
+   * @var array
+   */
+  protected $allowedDomainConfig = [
+    'system.site',
+    'system.theme*',
+    '*.theme.*',
+    '*.settings',
+    'node.settings',
+    'swiftmailer.*', // @TODO: Non-core items to be removed.
+  ];
+
+  /**
+   * The Domain config UI manager.
+   *
+   * @var Drupal\domain_config_ui\DomainConfigUIManager
+   */
+  protected $domainConfigUIManager;
+
+  /**
+   * Constructs the Config factory.
+   *
+   * @param \Drupal\Core\Config\StorageInterface $storage
+   *   The configuration storage engine.
+   * @param \Symfony\Component\EventDispatcher\EventDispatcherInterface $event_dispatcher
+   *   An event dispatcher instance to use for configuration events.
+   * @param \Drupal\Core\Config\TypedConfigManagerInterface $typed_config
+   *   The typed configuration manager.
+   * @param \Drupal\domain_config_ui\DomainConfigUIManager $domain_config_ui_manager
+   *   The domain config UI manager.
+   */
+  public function __construct(StorageInterface $storage,
+    EventDispatcherInterface $event_dispatcher,
+    TypedConfigManagerInterface $typed_config,
+    DomainConfigUIManager $domain_config_ui_manager) {
+    parent::__construct($storage, $event_dispatcher, $typed_config);
+    $this->domainConfigUIManager = $domain_config_ui_manager;
+  }
+
+  /**
+   * Helper to check if config is allowed to be saved for domain.
+   *
+   * @param string $name
+   *   The configuration name.
+   */
+  protected function isAllowedDomainConfig(string $name) {
```

Comment:
This is PHP7 only code, the string type hint should be removed

Reply:
Most type hints are supported but not 'string' as primitive type. That one is added in php 7.0, see http://php.net/manual/en/migration70.new-features.php

Reply:
BTW: I noticed this because I got a hard failure on my php 5.6 setup. So I'm 100% sure ;).

Reply:
Ah, was using PHP 7 for so long, I missed that. Fair comment.

Reply:
Yeah, we don't type hint strings anywhere else. But oddly this doesn't fail here -- https://travis-ci.org/agentrickard/domain/builds/398266990?utm_source=github_status&utm_medium=notification